docker/packager/binary-builder/Dockerfile

```diff
@@ -13,8 +13,8 @@ ENV CARGO_HOME=/rust/cargo
 ENV PATH="/rust/cargo/bin:${PATH}"
 RUN curl https://sh.rustup.rs -sSf | bash -s -- -y && \
     chmod 777 -R /rust && \
-    rustup toolchain install nightly-2024-12-01 && \
-    rustup default nightly-2024-12-01 && \
+    rustup toolchain install nightly-2024-04-01 && \
```

Comment:
Don't forget to revert this, rustup will be fixed in https://github.com/ClickHouse/ClickHouse/pull/77454

Reply:
Actually it will take a while I think, since there are some issues, so better to use older rustup (should not be that hard to install I guess)